src/gui/sharee.cpp

```diff
@@ -81,64 +81,26 @@ void ShareeModel::fetch(const QString &search, const ShareeSet &blacklist)
 
 void ShareeModel::shareesFetched(const QJsonDocument &reply)
 {
-    const auto data = reply.object().value("ocs").toObject().value("data").toObject();
-
     QVector<QSharedPointer<Sharee>> newSharees;
 
     /*
      * Todo properly loop all of this
      */
-    const auto exact = data.value("exact").toObject();
     {
-        const auto users = exact.value("users").toArray();
-        foreach (const auto &user, users) {
-            newSharees.append(parseSharee(user.toObject()));
-        }
-
-        const auto groups = exact.value("groups").toArray();
-        foreach (const auto &group, groups) {
-            newSharees.append(parseSharee(group.toObject()));
-        }
 
-        const auto emails = exact.value("emails").toArray();
-        foreach (const auto &email, emails) {
-            newSharees.append(parseSharee(email.toObject()));
-        }
+        const auto appendSharees = [this](const QJsonObject &data, QVector<QSharedPointer<Sharee>>& out) {
+            const QStringList shareeCategoryKeys {"users", "groups", "emails", "remotes", "circles"};
```

Comment:
I'd try to get it to be constexpr or I'd move it out of the lambda just to avoid allocating/deallocating it each time.

Writing the following might just work (but see my other comment below):
`constexpr auto shareeCategoryKeys = { "users", "groups", "emails", "remotes", "circles" };`

Reply:
@er-vin I'll move it out of lambda. Constexpr does not work with QString, unfortunately. So I either have to use array "const char*" (but then on each iteration "const char*" will get converted to QString when calling QJsonObject::QJsonValue value(, or move the list out of lambda.